frontend/coprs_frontend/coprs/views/apiv3_ns/apiv3_build_chroots.py

```diff
@@ -36,41 +55,89 @@ def build_config(build_chroot):
     return dict_data
 
 
-@apiv3_ns.route("/build-chroot", methods=GET)
-@apiv3_ns.route("/build-chroot/<int:build_id>/<chrootname>", methods=GET)  # deprecated
-@query_params()
-def get_build_chroot(build_id, chrootname):
-    chroot = ComplexLogic.get_build_chroot(build_id, chrootname)
-    return flask.jsonify(to_dict(chroot))
-
-
-@apiv3_ns.route("/build-chroot/list", methods=GET)
-@apiv3_ns.route("/build-chroot/list/<int:build_id>", methods=GET)
-@pagination()
-@query_params()
-def get_build_chroot_list(build_id, **kwargs):
-    # For the python3-copr <= 1.105
-    if kwargs.get("order") == "name":
-        kwargs.pop("order")
-    query = BuildChrootsLogic.filter_by_build_id(BuildChrootsLogic.get_multiply(), build_id)
-    paginator = Paginator(query, models.BuildChroot, **kwargs)
-    chroots = paginator.map(to_dict)
-    return flask.jsonify(items=chroots, meta=paginator.meta)
-
-
-@apiv3_ns.route("/build-chroot/build-config", methods=GET)
-@apiv3_ns.route("/build-chroot/build-config/<int:build_id>/<chrootname>", methods=GET)  # deprecated
-@query_params()
-def get_build_chroot_config(build_id, chrootname):
-    chroot = ComplexLogic.get_build_chroot(build_id, chrootname)
-    return flask.jsonify(build_config(chroot))
-
-
-@apiv3_ns.route("/build-chroot/built-packages/", methods=GET)
-@query_params()
-def get_build_chroot_built_packages(build_id, chrootname):
-    """
-    Return built packages (NEVRA dicts) for a given build chroot
-    """
-    chroot = ComplexLogic.get_build_chroot(build_id, chrootname)
-    return flask.jsonify(chroot.results_dict)
+@apiv3_bchroots_ns.route("/")
+@apiv3_bchroots_ns.route(
+    "/<int:build_id>/<chrootname>",
+    doc={"deprecated": True, "description": "Use query parameters instead"},
+)
+class BuildChroot(Resource):
+    @query_to_parameters
+    @apiv3_bchroots_ns.doc(params=build_chroot_params)
+    @apiv3_bchroots_ns.marshal_with(build_chroot_model)
+    @apiv3_bchroots_ns.response(HTTPStatus.OK.value, "OK, Build chroot data follows...")
+    @apiv3_bchroots_ns.response(
+        HTTPStatus.NOT_FOUND.value, "No such Build chroot exist"
+    )
+    def get(self, build_id, chrootname):
+        """
+        Get build chroot
+        Get information about specific build chroot by build id and mock chroot name.
+        """
+        chroot = ComplexLogic.get_build_chroot(build_id, chrootname)
+        return to_dict(chroot)
+
+
+@apiv3_bchroots_ns.route("/list")
+@apiv3_bchroots_ns.route("/list/<int:build_id>")  # TODO: ask if this is also deprecated?
```

Comment:
I saw deprecated comments to these routes but not in this one - should I also mark it as deprecated in documentation?

Reply:
I think it would make sense to mark this deprecated as well.

Reply:
done, PTAL